dbeam-core/src/main/java/com/spotify/dbeam/args/QueryBuilderArgs.java

```diff
@@ -98,23 +103,44 @@ private static Boolean checkTableName(String tableName) {
 
   public static QueryBuilderArgs create(String tableName) {
     checkArgument(tableName != null,
-        "TableName cannot be null");
-    checkArgument(checkTableName(tableName),
-        "'table' must follow [a-zA-Z_][a-zA-Z0-9_]*");
+            "TableName cannot be null");
+    DbeamQueryBuilder baseSqlQuery = getBaseSqlQuery(tableName, Optional.empty());
+    return new AutoValue_QueryBuilderArgs.Builder()
+            .setTableName(tableName)
+            .setBaseSqlQuery(baseSqlQuery)
+            .setPartitionPeriod(Days.ONE)
+            .build();
+  }
+
+  public static QueryBuilderArgs create(String tableName, final Optional<String> sqlQueryOpt) {
+    checkArgument(tableName != null,
+            "TableName cannot be null");
+    DbeamQueryBuilder baseSqlQuery = getBaseSqlQuery(tableName, sqlQueryOpt);
     return new AutoValue_QueryBuilderArgs.Builder()
         .setTableName(tableName)
+        .setBaseSqlQuery(baseSqlQuery)
         .setPartitionPeriod(Days.ONE)
         .build();
   }
 
+  private static DbeamQueryBuilder getBaseSqlQuery(String tableName, Optional<String> sqlQueryOpt) {
+    checkArgument(checkTableName(tableName), "'table' must follow [a-zA-Z_][a-zA-Z0-9_]*");
+    if (!sqlQueryOpt.isPresent()) {
```

Comment:
Would be nice to use `sqlQueryOpt.map` followed by `getOrElse` or something like that instead of `if`. No hard preference though. 
I think `if (sqlQueryOpt.isPresent())` is more readable than `if (!sqlQueryOpt.isPresent())` though.

Reply:
Changed.